rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -968,326 +1023,193 @@ object ExecutionBatch extends Loggable {
+              }
           }
         }
-        val status = buildComponentValueStatus(
-            unexpandedValue
-          , reports
-          , unexpectedStatusReports.isEmpty
-          , values.size
-          , noAnswerType
-          , policyMode
-        )
-        (status, reports)
       }
-      (result.keys,result.values.flatten)
-    }
 
-    // Remove all already parsed reports so we only look in remaining reports for Cfengine variables
-    val usedReports = noneReports ++ simpleReports ++ unexpectedReports
-    val remainingReports = filteredReports.diff(usedReports)
+      reports match {
+        case Nil            => (freeValues ::: pairedValues, unexpected)
+        case report :: tail =>
+          // we look in the free values for one matching the report. If found, we return (remaining freevalues, Some[paired value]
+          // else (all free values, None).
+          // never increment the cardinality for free value.
+          val (newFreeValues, tryPair) = findMachingValue(report, freeValues, (value, report) => false, (value, report) => false)
+
+          logger.trace(s"found unpaired value for '${report.keyValue}'? " + tryPair)
+
+          tryPair match {
+            case Some(v) =>
+              recPairReports(tail, newFreeValues, v :: pairedValues, unexpected, mode)
+            case None    =>
+              // here, we don't have found any free value for that report. We are not done yet because it can be an
+              // unexpected reprots bound to an already existing value (and so the whole component should be
+              // unexpected or if mode allows duplicates or unbound var values, it can be ok.
+
+              val duplicate = (value: Value, report: ResultReports) => {
+                mode.isSet(UnexpectedReportBehavior.AllowsDuplicate) && {
+                  // for duplicate, we want exact same report than already accepted (it must be a real duplicate)
+                  // and we will also forbid more than 3 duplicates for the same component value. Because if there is
+                  // more than 3, it's OK to raise attention of people on that, it may be an other problem than syslog
+                  // being made, or syslog being so made than something must be done.
+                  val dup = value.matchingReports.collect { case r if(r == report) => r}
+                  //predicate OK if we found at least one identical report
+                  dup.size >= 1
+                }
+              }
+              val unboundedVar = (value: Value, report: ResultReports) => {
+                mode.isSet(UnexpectedReportBehavior.UnboundVarValues) && {
+                  // this predicate is simpler: it just have to be a variable
+                  value.isVar
+                }
+              }
 
-    // Find what reports matche what cfengine variables
+              val (newPairedValues, pairedAgain) = findMachingValue(report, pairedValues, duplicate, unboundedVar)
```

Comment:
No, we need that to know what is the duplicated value and set the "unexpected" accordingly. 
(it may also happen that none match again, but it is an other case).